Add Tag Extension structures

glamberson opened the pull request:
## Summary
This PR adds structures for the GEDCOM Tag Extension, which enables preservation of organizational tags commonly used in genealogy software.

## Extension Overview
- **Purpose**: Preserve user-defined tags with colors and vendor-specific attributes
- **Repository**: https://github.com/glamberson/gedcom-tags
- **Version**: 0.1.0
- **Contact**: glamberson@gmail.com

## Structures Added
- `_TAG` - Tag record type for defining reusable organizational markers
- `_TAG` (ref) - Substructure for applying tags to records  
- `_COLOR` - Optional color specification (6-digit hex)
- `_ATTR` - Generic vendor attribute mechanism
- `_VALUE` - Value for generic attributes

## Key Features
1. Universal application to any record type
2. Optional visual color coding  
3. Vendor extensibility through _ATTR
4. Round-trip preservation of tag data

## Use Case
Modern genealogy software universally implements tagging/categorization systems, but these are lost during GEDCOM export. This extension preserves:
- Research status tags (ToDo, InProgress, Complete)
- DNA match markers
- Visual color coding
- Software-specific tag attributes

## Documentation
Full specification and examples available at: https://github.com/glamberson/gedcom-tags

## Testing
Example GEDCOM files demonstrating usage are included in the repository:
- [basic.ged](https://github.com/glamberson/gedcom-tags/blob/main/examples/basic.ged) - Simple tag usage
- [advanced.ged](https://github.com/glamberson/gedcom-tags/blob/main/examples/advanced.ged) - Vendor attributes
- [gramps-compatibility.ged](https://github.com/glamberson/gedcom-tags/blob/main/examples/gramps-compatibility.ged) - Hierarchical tags

glamberson commented:
Thank you for pointing out the prerelease field\! 

We do not need `prerelease: true` for this extension. We consider the syntax stable and immutable at this point. The Tag extension has been:

- Released as v0.1.0 with semantic versioning
- Actively tested with GRAMPS implementation (PR https://github.com/DavidMStraub/gramps-gedcom7/pull/1)
- Used in production test files

We're committed to maintaining backward compatibility going forward. Any future enhancements will follow semantic versioning principles.

Also, I've just pushed fixes for the YAML validation errors - moving files to the proper structure/extension/ subdirectory and adding the extension manifest file.

glamberson commented:
Fixed YAML validation errors - added missing newlines at the end of all YAML files. Also tagged v0.2.2.

tychonievich commented on extension-tags.yaml:
```diff
@@ -0,0 +1,29 @@
+  - https://github.com/glamberson/gedcom-tags/_VALUE
+
+contact: lamberson@yahoo.com
+...
```
```suggestion
...

```

tychonievich commented on structure/extension/_ATTR.yaml:
```diff
@@ -0,0 +1,34 @@
+  "https://github.com/glamberson/gedcom-tags/_TAG": "{0:M}"
+
+contact: "lamberson@yahoo.com"
+...
```
```suggestion
...

```

tychonievich commented on structure/extension/_COLOR.yaml:
```diff
@@ -0,0 +1,33 @@
+  "https://github.com/glamberson/gedcom-tags/_TAG": "{0:1}"
+
+contact: "lamberson@yahoo.com"
+...
```
```suggestion
...

```

tychonievich commented on structure/extension/_TAG.yaml:
```diff
@@ -0,0 +1,39 @@
+superstructures: {}
+
+contact: "lamberson@yahoo.com"
+...
```
```suggestion
...

```

tychonievich commented on structure/extension/_TAG_ref.yaml:
```diff
@@ -0,0 +1,48 @@
+  "https://gedcom.io/terms/v7/ASSO": "{0:M}"
+
+contact: "lamberson@yahoo.com"
+...
```
```suggestion
...

```

tychonievich commented on structure/extension/_VALUE.yaml:
```diff
@@ -0,0 +1,30 @@
+  "https://github.com/glamberson/gedcom-tags/_ATTR": "{1:1}"
+
+contact: "lamberson@yahoo.com"
+...
```
```suggestion
...

```

tychonievich reviewed (commented):
YAML validation fix: each YAML file must end with a newline character.

tychonievich commented on extension-tags.yaml:
```diff
@@ -0,0 +1,29 @@
+%YAML 1.2
```
An issue to track registering multi-part extensions has been created in #179

dthaler commented on structure/extension/_ATTR.yaml:
```diff
@@ -0,0 +1,34 @@
+
+uri: https://github.com/glamberson/gedcom-tags/_ATTR
+
+standard tag: _ATTR
```
Don't use "standard tag", its for things that have already been standardized.  Just use "extension tags"

dthaler commented on structure/extension/_TAG_ref.yaml:
```diff
@@ -0,0 +1,48 @@
+
+type: structure
+
+uri: https://github.com/glamberson/gedcom-tags/_TAG
```
This must have a separate URI from the one in structure/extension/_TAG.yaml

dthaler commented on structure/extension/_TAG_ref.yaml:
```diff
@@ -0,0 +1,48 @@
+
+label: 'Tag Reference'
+
+payload: https://gedcom.io/terms/v7/type-Pointer
```
```suggestion
payload: "@<https://github.com/glamberson/gedcom-tags/_TAG>@"
```

dthaler commented on structure/extension/_ATTR.yaml:
```diff
@@ -0,0 +1,32 @@
+
+uri: https://github.com/glamberson/gedcom-tags/_ATTR
+
+extension tags: _ATTR
```
```suggestion
extension tags:
  - _ATTR
```

glamberson commented:
@dthaler @tychonievich I've addressed all review feedback in commit f94bc01:

- Removed 'standard tag' fields (only using 'extension tags')
- Fixed array syntax for extension tags  
- Added blank lines before closing '...'
- Made _TAG_ref URI unique (different from _TAG)
- Fixed payload format to proper pointer syntax
- Removed extension-tags.yaml file

Ready for re-review.

dthaler commented on structure/extension/_ATTR.yaml:
```diff
@@ -0,0 +1,34 @@
+    
+    This allows software to extend tags with specific functionality
+    without requiring changes to the base specification.
+
```
Recommend adding
```
documentation:
  - https://github.com/glamberson/gedcom-tags
```
To YAML files in this PR.

dthaler commented on structure/extension/_TAG_ref.yaml:
```diff
@@ -0,0 +1,48 @@
+  "https://gedcom.io/terms/v7/ASSO": "{0:M}"
+
+contact: "lamberson@yahoo.com"
+
```
if this is in use by gramps as you said:
> Actively tested with GRAMPS implementation

Then add `used by` as well to these YAML files.
